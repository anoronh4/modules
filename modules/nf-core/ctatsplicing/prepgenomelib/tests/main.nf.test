nextflow_process {

    name "Test Process CTATSPLICING_PREPGENOMELIB"
    script "../main.nf"
    process "CTATSPLICING_PREPGENOMELIB"

    tag "modules"
    tag "modules_nfcore"
    tag "ctatsplicing"
    tag "ctatsplicing/prepgenomelib"

    test("homo_sapiens - reference - stub") {

        options "-stub"

        when {
            process {
                """
                new File('ctat_genome_lib_build_dir').mkdirs()
                input[0] = [[id:"ctat_genome_lib_build_dir"],file("ctat_genome_lib_build_dir/")]
                input[1] = [
                    file("https://data.broadinstitute.org/Trinity/CTAT_RESOURCE_LIB/CANCER_SPLICING_LIB_SUPPLEMENT/cancer_introns.GRCh38.Jun232020.tsv.gz", checkIfExists:true)
                ]
                """
            }
        }

        then {
            def stable_name    = getAllFilesFromDir(process.out.reference.get(0).get(1), includeDir: true)
            def stable_content = getAllFilesFromDir(process.out.reference.get(0).get(1), includeDir: false, include: ['*','**/*'], ignore: ['refGene.sort.bed.gz.tbi','**/refGene.sort.bed.gz.tbi','**/cancer_splicing.idx'])
            assertAll(
                { assert process.success },
                {
                    assert snapshot(
                        stable_name*.name,
                        stable_content,
                        process.out.versions
                    ).match()
                }
            )

        }

    }

}
