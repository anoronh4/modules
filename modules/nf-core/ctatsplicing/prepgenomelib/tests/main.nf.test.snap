{
    "homo_sapiens - reference - stub": {
        "content": [
            [
                "ctat_genome_lib_build_dir",
                "AnnotFilterRule.pm",
                "PFAM.domtblout.dat.gz",
                "__chkpts",
                "_fusion_annot_lib.idx.ok",
                "_prot_info_db.ok",
                "annotfiltrule_cp.ok",
                "blast_pairs.idx.ok",
                "cp_gene_blast_pairs.ok",
                "cp_pfam_dat.ok",
                "cp_ref_annot_cdna.ok",
                "fusion_annot_lib.cp.ok",
                "index_pfam_hits.ok",
                "index_ref_annot_cdna.ok",
                "makeblastdb.ok",
                "mm2.splice_bed.ok",
                "mm2_genome_idx.ok",
                "ref_annot.cdsplus.dfam_masked.fa.cp.ok",
                "ref_annot.cdsplus.dfam_masked.fa.idx.ok",
                "ref_annot.gtf.gene_spans.ok",
                "ref_annot.gtf.mini.sortu.ok",
                "ref_annot.gtf.ok",
                "ref_genome.fa.ok",
                "ref_genome_fai.ok",
                "trans.blast.dat.cp.ok",
                "trans.blast.dat.index.ok",
                "validate_ctat_genome_lib.ok",
                "blast_pairs.dat.gz",
                "blast_pairs.idx",
                "cancer_splicing_lib",
                "cancer_splicing.idx",
                "fusion_annot_lib.gz",
                "fusion_annot_lib.idx",
                "pfam_domains.dbm",
                "refGene.bed",
                "refGene.sort.bed.gz",
                "refGene.sort.bed.gz.tbi",
                "ref_annot.cdna.fa",
                "ref_annot.cdna.fa.idx",
                "ref_annot.cds",
                "ref_annot.cdsplus.fa",
                "ref_annot.cdsplus.fa.idx",
                "ref_annot.gtf",
                "ref_annot.gtf.gene_spans",
                "ref_annot.gtf.mini.sortu",
                "ref_annot.gtf.mm2.splice.bed",
                "ref_annot.pep",
                "ref_annot.prot_info.dbm",
                "ref_genome.fa",
                "ref_genome.fa.fai",
                "ref_genome.fa.mm2",
                "ref_genome.fa.ndb",
                "ref_genome.fa.nhr",
                "ref_genome.fa.nin",
                "ref_genome.fa.njs",
                "ref_genome.fa.not",
                "ref_genome.fa.nsq",
                "ref_genome.fa.ntf",
                "ref_genome.fa.nto",
                "ref_genome.fa.star.idx",
                "Genome",
                "Log.out",
                "SA",
                "SAindex",
                "build.ok",
                "chrLength.txt",
                "chrName.txt",
                "chrNameLength.txt",
                "chrStart.txt",
                "exonGeTrInfo.tab",
                "exonInfo.tab",
                "geneInfo.tab",
                "genomeParameters.txt",
                "sjdbInfo.txt",
                "sjdbList.fromGTF.out.tab",
                "sjdbList.out.tab",
                "transcriptInfo.tab",
                "trans.blast.align_coords.align_coords.dat",
                "trans.blast.align_coords.align_coords.dbm",
                "trans.blast.dat.gz"
            ],
            [
                "AnnotFilterRule.pm:md5,d41d8cd98f00b204e9800998ecf8427e",
                "PFAM.domtblout.dat.gz:md5,68b329da9893e34099c7d8ad5cb9c940",
                "_fusion_annot_lib.idx.ok:md5,d41d8cd98f00b204e9800998ecf8427e",
                "_prot_info_db.ok:md5,d41d8cd98f00b204e9800998ecf8427e",
                "annotfiltrule_cp.ok:md5,d41d8cd98f00b204e9800998ecf8427e",
                "blast_pairs.idx.ok:md5,d41d8cd98f00b204e9800998ecf8427e",
                "cp_gene_blast_pairs.ok:md5,d41d8cd98f00b204e9800998ecf8427e",
                "cp_pfam_dat.ok:md5,d41d8cd98f00b204e9800998ecf8427e",
                "cp_ref_annot_cdna.ok:md5,d41d8cd98f00b204e9800998ecf8427e",
                "fusion_annot_lib.cp.ok:md5,d41d8cd98f00b204e9800998ecf8427e",
                "index_pfam_hits.ok:md5,d41d8cd98f00b204e9800998ecf8427e",
                "index_ref_annot_cdna.ok:md5,d41d8cd98f00b204e9800998ecf8427e",
                "makeblastdb.ok:md5,d41d8cd98f00b204e9800998ecf8427e",
                "mm2.splice_bed.ok:md5,d41d8cd98f00b204e9800998ecf8427e",
                "mm2_genome_idx.ok:md5,d41d8cd98f00b204e9800998ecf8427e",
                "ref_annot.cdsplus.dfam_masked.fa.cp.ok:md5,d41d8cd98f00b204e9800998ecf8427e",
                "ref_annot.cdsplus.dfam_masked.fa.idx.ok:md5,d41d8cd98f00b204e9800998ecf8427e",
                "ref_annot.gtf.gene_spans.ok:md5,d41d8cd98f00b204e9800998ecf8427e",
                "ref_annot.gtf.mini.sortu.ok:md5,d41d8cd98f00b204e9800998ecf8427e",
                "ref_annot.gtf.ok:md5,d41d8cd98f00b204e9800998ecf8427e",
                "ref_genome.fa.ok:md5,d41d8cd98f00b204e9800998ecf8427e",
                "ref_genome_fai.ok:md5,d41d8cd98f00b204e9800998ecf8427e",
                "trans.blast.dat.cp.ok:md5,d41d8cd98f00b204e9800998ecf8427e",
                "trans.blast.dat.index.ok:md5,d41d8cd98f00b204e9800998ecf8427e",
                "validate_ctat_genome_lib.ok:md5,d41d8cd98f00b204e9800998ecf8427e",
                "blast_pairs.dat.gz:md5,68b329da9893e34099c7d8ad5cb9c940",
                "blast_pairs.idx:md5,d41d8cd98f00b204e9800998ecf8427e",
                "fusion_annot_lib.gz:md5,68b329da9893e34099c7d8ad5cb9c940",
                "fusion_annot_lib.idx:md5,d41d8cd98f00b204e9800998ecf8427e",
                "pfam_domains.dbm:md5,d41d8cd98f00b204e9800998ecf8427e",
                "refGene.bed:md5,d41d8cd98f00b204e9800998ecf8427e",
                "refGene.sort.bed.gz:md5,68b329da9893e34099c7d8ad5cb9c940",
                "ref_annot.cdna.fa:md5,d41d8cd98f00b204e9800998ecf8427e",
                "ref_annot.cdna.fa.idx:md5,d41d8cd98f00b204e9800998ecf8427e",
                "ref_annot.cds:md5,d41d8cd98f00b204e9800998ecf8427e",
                "ref_annot.cdsplus.fa:md5,d41d8cd98f00b204e9800998ecf8427e",
                "ref_annot.cdsplus.fa.idx:md5,d41d8cd98f00b204e9800998ecf8427e",
                "ref_annot.gtf:md5,d41d8cd98f00b204e9800998ecf8427e",
                "ref_annot.gtf.gene_spans:md5,d41d8cd98f00b204e9800998ecf8427e",
                "ref_annot.gtf.mini.sortu:md5,d41d8cd98f00b204e9800998ecf8427e",
                "ref_annot.gtf.mm2.splice.bed:md5,d41d8cd98f00b204e9800998ecf8427e",
                "ref_annot.pep:md5,d41d8cd98f00b204e9800998ecf8427e",
                "ref_annot.prot_info.dbm:md5,d41d8cd98f00b204e9800998ecf8427e",
                "ref_genome.fa:md5,d41d8cd98f00b204e9800998ecf8427e",
                "ref_genome.fa.fai:md5,d41d8cd98f00b204e9800998ecf8427e",
                "ref_genome.fa.mm2:md5,d41d8cd98f00b204e9800998ecf8427e",
                "ref_genome.fa.ndb:md5,d41d8cd98f00b204e9800998ecf8427e",
                "ref_genome.fa.nhr:md5,d41d8cd98f00b204e9800998ecf8427e",
                "ref_genome.fa.nin:md5,d41d8cd98f00b204e9800998ecf8427e",
                "ref_genome.fa.njs:md5,d41d8cd98f00b204e9800998ecf8427e",
                "ref_genome.fa.not:md5,d41d8cd98f00b204e9800998ecf8427e",
                "ref_genome.fa.nsq:md5,d41d8cd98f00b204e9800998ecf8427e",
                "ref_genome.fa.ntf:md5,d41d8cd98f00b204e9800998ecf8427e",
                "ref_genome.fa.nto:md5,d41d8cd98f00b204e9800998ecf8427e",
                "Genome:md5,d41d8cd98f00b204e9800998ecf8427e",
                "Log.out:md5,d41d8cd98f00b204e9800998ecf8427e",
                "SA:md5,d41d8cd98f00b204e9800998ecf8427e",
                "SAindex:md5,d41d8cd98f00b204e9800998ecf8427e",
                "build.ok:md5,d41d8cd98f00b204e9800998ecf8427e",
                "chrLength.txt:md5,d41d8cd98f00b204e9800998ecf8427e",
                "chrName.txt:md5,d41d8cd98f00b204e9800998ecf8427e",
                "chrNameLength.txt:md5,d41d8cd98f00b204e9800998ecf8427e",
                "chrStart.txt:md5,d41d8cd98f00b204e9800998ecf8427e",
                "exonGeTrInfo.tab:md5,d41d8cd98f00b204e9800998ecf8427e",
                "exonInfo.tab:md5,d41d8cd98f00b204e9800998ecf8427e",
                "geneInfo.tab:md5,d41d8cd98f00b204e9800998ecf8427e",
                "genomeParameters.txt:md5,d41d8cd98f00b204e9800998ecf8427e",
                "sjdbInfo.txt:md5,d41d8cd98f00b204e9800998ecf8427e",
                "sjdbList.fromGTF.out.tab:md5,d41d8cd98f00b204e9800998ecf8427e",
                "sjdbList.out.tab:md5,d41d8cd98f00b204e9800998ecf8427e",
                "transcriptInfo.tab:md5,d41d8cd98f00b204e9800998ecf8427e",
                "trans.blast.align_coords.align_coords.dat:md5,d41d8cd98f00b204e9800998ecf8427e",
                "trans.blast.align_coords.align_coords.dbm:md5,d41d8cd98f00b204e9800998ecf8427e",
                "trans.blast.dat.gz:md5,68b329da9893e34099c7d8ad5cb9c940"
            ],
            [
                "versions.yml:md5,3554c611ce8fa5c21460b6a99e8e36dc"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.3"
        },
        "timestamp": "2025-03-31T16:20:05.986978"
    }
}